### ISSUE TITLE
[BUG] SKSurface.Snapshot() returns #00000000 pixels on UWP

### ISSUE THREAD

**Issue by daltonks** [mono/SkiaSharp#902]
**Description**

Clearing the canvas with `SKColors.White` then calling `SKSurface.Snapshot()` returns all #00000000 pixels on UWP. However, it works properly on Android.

**Code**

A solution that reproduces the issue can be found [here](https://github.com/daltonks/SkiaSharp.UwpSnapshotBug).

The main part is in the [code-behind](https://github.com/daltonks/SkiaSharp.UwpSnapshotBug/blob/master/SkiaSharp.UwpSnapshotBug/MainPage.xaml.cs) and the [xaml](https://github.com/daltonks/SkiaSharp.UwpSnapshotBug/blob/master/SkiaSharp.UwpSnapshotBug/MainPage.xaml).

**Expected Behavior**

The sample apps both write `Has non-zero color: True` to the console.

**Actual Behavior**

Android writes `Has non-zero color: True`, but UWP writes `Has non-zero color: False`

**Basic Information**

- Version with issue:  1.68.0
- Last known good version:  Not sure
- IDE:  Visual Studio
- Platform Target Frameworks: 
  - Android: 9.0
  - UWP:  16299 

- Target Devices:
  - Windows 10.0.18362
  - Android Pie on a Samsung Galaxy S8

**Comment by daltonks** [comment]
SKSurface.Snapshot() is returning a null image on iOS using this same method. I can create a new issue if needed.

**Comment by aodpi** [comment]
I also noticed that the `SKSurface.Snapshot()` returns null only when using `SKGLView`.

**Comment by Gillibald** [comment]
What hardware are you using? Is there a dedicated GPU?

**Comment by aodpi** [comment]
@Gillibald No, there is no dedicated gpu. It was a mac mini late 2014, macOS Mojave 10.14.5

**Comment by Gillibald** [comment]
I guess in case no dedicated GPU is available SKGLView can't be used.

**Comment by aodpi** [comment]
@Gillibald I guess, but in this case, shouldn't it fallback to something else ?

**Comment by Gillibald** [comment]
How do you emulate OpenGL HW-acceleration in software?

**Comment by aodpi** [comment]
@Gillibald I believe a good solution here would be to use `SKCanvasView` on UWP and `SKGLView` on Android, right ?

**Comment by Gillibald** [comment]
SKGLView should work on UWP with hardware support.
